src/com/sun/jna/NativeLibrary.java

```diff
@@ -132,7 +138,7 @@ private static NativeLibrary loadLibrary(String libraryName, Map options) {
 	}
 
         boolean isAbsolutePath = new File(libraryName).isAbsolute();
-        List searchPath = new LinkedList();
+        List<String> searchPath = new LinkedList<String>();
```

Comment:
Performance here is kind of a moot point, these things will never have more than a few entries.